SwiftLint violations in RPCServers, CcipRead

1) alpha-wallet-ios/modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/RPCServers.swift:88:360: Comma Spacing Violation: There should be no space before and one after any comma. (comma)

2) alpha-wallet-ios/modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/RPCServers.swift:1009:1: warning: File Line Length Violation: File should contain 1000 lines or less: currently contains 1009 (file_length)

3) alpha-wallet-ios/modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/RPCServers.swift:961:2: warning: Switch and Case Statement Alignment Violation: Case statements should vertically align with their enclosing switch statement. (switch_case_alignment)

4) alpha-wallet-ios/modules/AlphaWalletWeb3/AlphaWalletWeb3/Utils/CcipRead.swift:117:80: Large Tuple Violation: Tuples should have at most 4 members. (large_tuple)

5) alpha-wallet-ios/modules/AlphaWalletWeb3/AlphaWalletWeb3/Utils/CcipRead.swift:134:65: warning: Large Tuple Violation: Tuples should have at most 4 members. (large_tuple)

--- a/AlphaWallet/Localization/es.lproj/Localizable.strings
+++ b/AlphaWallet/Localization/es.lproj/Localizable.strings
@@ -467,6 +467,7 @@
 "blockchain.Mumbai" = "Mumbai Testnet";
 "blockchain.Optimistic" = "Optimistic Testnet";
 "blockchain.Optimistic.Kovan" = "Optimistic Kovan Testnet";
+"blockchain.Cronos.Mainnet" = "Cronos";
 "blockchain.Cronos.Testnet" = "Cronos Testnet";
 "blockchain.arbitrum" = "Arbitrum";
 "blockchain.arbitrum.rinkeby" = "Arbitrum Rinkeby";

--- a/AlphaWallet/Localization/fi.lproj/Localizable.strings
+++ b/AlphaWallet/Localization/fi.lproj/Localizable.strings
@@ -467,6 +467,7 @@
 "blockchain.Mumbai" = "Mumbai Testnet";
 "blockchain.Optimistic" = "Optimistic Testnet";
 "blockchain.Optimistic.Kovan" = "Optimistic Kovan Testnet";
+"blockchain.Cronos.Mainnet" = "Cronos";
 "blockchain.Cronos.Testnet" = "Cronos Testnet";
 "blockchain.arbitrum" = "Arbitrum";
 "blockchain.arbitrum.rinkeby" = "Arbitrum Rinkeby";

--- a/AlphaWallet/Localization/ja.lproj/Localizable.strings
+++ b/AlphaWallet/Localization/ja.lproj/Localizable.strings
@@ -467,6 +467,7 @@
 "blockchain.Mumbai" = "Mumbai Testnet";
 "blockchain.Optimistic" = "Optimistic Testnet";
 "blockchain.Optimistic.Kovan" = "Optimistic Kovan Testnet";
+"blockchain.Cronos.Mainnet" = "Cronos";
 "blockchain.Cronos.Testnet" = "Cronos Testnet";
 "blockchain.arbitrum" = "Arbitrum";
 "blockchain.arbitrum.rinkeby" = "Arbitrum Rinkeby";

--- a/AlphaWallet/Localization/ko.lproj/Localizable.strings
+++ b/AlphaWallet/Localization/ko.lproj/Localizable.strings
@@ -464,6 +464,7 @@
 "blockchain.Mumbai" = "Mumbai Testnet";
 "blockchain.Optimistic" = "Optimistic Testnet";
 "blockchain.Optimistic.Kovan" = "Optimistic Kovan Testnet";
+"blockchain.Cronos.Mainnet" = "Cronos";
 "blockchain.Cronos.Testnet" = "Cronos Testnet";
 "blockchain.arbitrum" = "Arbitrum";
 "blockchain.arbitrum.rinkeby" = "Arbitrum Rinkeby";

--- a/AlphaWallet/Localization/zh-Hans.lproj/Localizable.strings
+++ b/AlphaWallet/Localization/zh-Hans.lproj/Localizable.strings
@@ -467,6 +467,7 @@
 "blockchain.Mumbai" = "Mumbai Testnet";
 "blockchain.Optimistic" = "Optimistic Testnet";
 "blockchain.Optimistic.Kovan" = "Optimistic Kovan Testnet";
+"blockchain.Cronos.Mainnet" = "Cronos";
 "blockchain.Cronos.Testnet" = "Cronos Testnet";
 "blockchain.arbitrum" = "Arbitrum";
 "blockchain.arbitrum.rinkeby" = "Arbitrum Rinkeby";

--- a/AlphaWallet/Tokens/ViewModels/TokensViewModel.swift
+++ b/AlphaWallet/Tokens/ViewModels/TokensViewModel.swift
@@ -283,7 +283,6 @@ final class TokensViewModel {
         }.eraseToAnyPublisher()
     }
 
-
     private var isFooterHidden: Bool {
         !config.enabledServers.contains(.main)
     }

--- a/AlphaWallet/Transfer/ViewModels/SendViewModel.swift
+++ b/AlphaWallet/Transfer/ViewModels/SendViewModel.swift
@@ -127,7 +127,6 @@ final class SendViewModel {
             .eraseToAnyPublisher()
 
         let recipientErrorState = isRecipientValid(inputsValidationError: inputsValidationError)
-            .print("xxx.recipientErrorState")
             .map { $0 ? TextField.TextFieldErrorState.none : TextField.TextFieldErrorState.error(InputError.invalidAddress.prettyError) }
             .eraseToAnyPublisher()
 

--- a/AlphaWalletTests/Core/Types/AddressStorageTests.swift
+++ b/AlphaWalletTests/Core/Types/AddressStorageTests.swift
@@ -29,30 +29,30 @@ private struct FakeTickerId: Codable {
 class AddressStorageTests: XCTestCase {
 
     func testInMemoryAddressStorage() throws {
-        return; //NOTE: disable this test as it takes too mutch time for execution
-
-        guard let bundlePath = Bundle(for: AddressStorageTests.self).path(forResource: "tikersForTest", ofType: "json") else { XCTFail(); return }
-        guard let jsonData = try String(contentsOfFile: bundlePath).data(using: .utf8) else { XCTFail(); return }
-
-        self.measure {
-            let storage = InMemoryAddressStorage()
-            register(addressStorage: storage)
-
-            do {
-                _ = try JSONDecoder().decode([FakeTickerId].self, from: jsonData)
-            } catch {
-                XCTFail()
-            }
-
-            let storage2 = InMemoryAddressStorage()
-            register(addressStorage: storage2)
-
-            do {
-                _ = try JSONDecoder().decode([FakeTickerId].self, from: jsonData)
-            } catch {
-                XCTFail()
-            }
-        }
+//        return; //NOTE: disable this test as it takes too mutch time for execution
+//
+//        guard let bundlePath = Bundle(for: AddressStorageTests.self).path(forResource: "tikersForTest", ofType: "json") else { XCTFail() }
+//        guard let jsonData = try String(contentsOfFile: bundlePath).data(using: .utf8) else { XCTFail(); return }
+//
+//        self.measure {
+//            let storage = InMemoryAddressStorage()
+//            register(addressStorage: storage)
+//
+//            do {
+//                _ = try JSONDecoder().decode([FakeTickerId].self, from: jsonData)
+//            } catch {
+//                XCTFail()
+//            }
+//
+//            let storage2 = InMemoryAddressStorage()
+//            register(addressStorage: storage2)
+//
+//            do {
+//                _ = try JSONDecoder().decode([FakeTickerId].self, from: jsonData)
+//            } catch {
+//                XCTFail()
+//            }
+//        }
     }
 
     func testInFileAddressStorage() throws {

--- a/modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/RPCServers.swift
+++ b/modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/RPCServers.swift
@@ -85,7 +85,7 @@ public enum RPCServer: Hashable, CaseIterable {
 
     public var isDeprecated: Bool {
         switch self {
-        case .xDai, .phi, .phi2, .classic, .main, .poa, .callisto, .binance_smart_chain, .artis_sigma1, .heco, .fantom, .avalanche, .candle, .polygon, .optimistic, .arbitrum, .palm, .klaytnCypress, .ioTeX, .goerli, .artis_tau1, .binance_smart_chain_testnet, .heco_testnet, .fantom_testnet, .avalanche_testnet, .mumbai_testnet, .optimisticKovan, .cronosMainnet,  .cronosTestnet, .palmTestnet, .arbitrumRinkeby, .klaytnBaobabTestnet, .ioTeXTestnet, .custom, .optimismGoerli, .arbitrumGoerli:
+        case .xDai, .phi, .phi2, .classic, .main, .poa, .callisto, .binance_smart_chain, .artis_sigma1, .heco, .fantom, .avalanche, .candle, .polygon, .optimistic, .arbitrum, .palm, .klaytnCypress, .ioTeX, .goerli, .artis_tau1, .binance_smart_chain_testnet, .heco_testnet, .fantom_testnet, .avalanche_testnet, .mumbai_testnet, .optimisticKovan, .cronosMainnet, .cronosTestnet, .palmTestnet, .arbitrumRinkeby, .klaytnBaobabTestnet, .ioTeXTestnet, .custom, .optimismGoerli, .arbitrumGoerli:
             return false
         case .kovan, .ropsten, .rinkeby, .sokol: //NOTE: maybe Arbitrum Kovan, Optimistic Rinkeby too
             return true
@@ -877,44 +877,28 @@ public enum RPCServer: Hashable, CaseIterable {
 
     public var explorerName: String {
         switch self {
-        case .main, .kovan, .ropsten, .rinkeby, .goerli, .optimismGoerli, .arbitrumGoerli:
-            return "Etherscan"
-        case .classic, .poa, .custom, .callisto, .sokol, .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .candle, .polygon, .mumbai_testnet, .optimistic, .optimisticKovan, .arbitrum, .arbitrumRinkeby, .palm, .palmTestnet, .klaytnCypress, .klaytnBaobabTestnet, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli:
-            return "\(name) Explorer"
-        case .xDai:
-            return "Blockscout"
-        case .artis_sigma1, .artis_tau1:
-            return "ARTIS"
-        case .cronosMainnet, .cronosTestnet:
-            return "Cronoscan"
+        case .main, .kovan, .ropsten, .rinkeby, .goerli, .optimismGoerli, .arbitrumGoerli: return "Etherscan"
+        case .classic, .poa, .custom, .callisto, .sokol, .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .candle, .polygon, .mumbai_testnet, .optimistic, .optimisticKovan, .arbitrum, .arbitrumRinkeby, .palm, .palmTestnet, .klaytnCypress, .klaytnBaobabTestnet, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli: return "\(name) Explorer"
+        case .xDai: return "Blockscout"
+        case .artis_sigma1, .artis_tau1: return "ARTIS"
+        case .cronosMainnet, .cronosTestnet: return "Cronoscan"
         }
     }
 
     //Implementation: Almost every chain should return nil here
     public var serverWithEnhancedSupport: RPCServerWithEnhancedSupport? {
         switch self {
-        case .main:
-            return .main
-        case .xDai:
-            return .xDai
-        case .candle:
-            return .candle
-        case .polygon:
-            return .polygon
-        case .binance_smart_chain:
-            return .binance_smart_chain
-        case .heco:
-            return .heco
-        case .rinkeby:
-            return .rinkeby
-        case .arbitrum:
-            return .arbitrum
-        case .klaytnCypress:
-            return .klaytnCypress
-        case .klaytnBaobabTestnet:
-            return .klaytnBaobabTestnet
-        case .main, .kovan, .ropsten, .sokol, .goerli, .artis_sigma1, .artis_tau1, .custom, .poa, .callisto, .xDai, .classic, .binance_smart_chain_testnet, .heco_testnet, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .mumbai_testnet, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli, .cronosMainnet:
-            return nil
+        case .main: return .main
+        case .xDai: return .xDai
+        case .candle: return .candle
+        case .polygon: return .polygon
+        case .binance_smart_chain: return .binance_smart_chain
+        case .heco: return .heco
+        case .rinkeby: return .rinkeby
+        case .arbitrum: return .arbitrum
+        case .klaytnCypress: return .klaytnCypress
+        case .klaytnBaobabTestnet: return .klaytnBaobabTestnet
+        case .main, .kovan, .ropsten, .sokol, .goerli, .artis_sigma1, .artis_tau1, .custom, .poa, .callisto, .xDai, .classic, .binance_smart_chain_testnet, .heco_testnet, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .mumbai_testnet, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli, .cronosMainnet: return nil
         }
     }
 
@@ -930,38 +914,30 @@ public enum RPCServer: Hashable, CaseIterable {
         case .arbitrum: return "arbitrum-one"
         case .klaytnCypress, .klaytnBaobabTestnet: return "klay-token"
         case .cronosMainnet: return "cronos"
-        case .poa, .kovan, .sokol, .callisto, .goerli, .artis_sigma1, .artis_tau1, .binance_smart_chain_testnet, .ropsten, .rinkeby, .heco, .heco_testnet, .fantom_testnet, .avalanche_testnet, .candle, .mumbai_testnet, .custom, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli:
-            return nil
+        case .poa, .kovan, .sokol, .callisto, .goerli, .artis_sigma1, .artis_tau1, .binance_smart_chain_testnet, .ropsten, .rinkeby, .heco, .heco_testnet, .fantom_testnet, .avalanche_testnet, .candle, .mumbai_testnet, .custom, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli: return nil
         }
     }
 
     var coinbasePlatform: String? {
         switch self {
         case .main: return "ethereum"
-        case .avalanche, .xDai, .classic, .fantom, .arbitrum, .candle, .polygon, .binance_smart_chain, .klaytnCypress, .klaytnBaobabTestnet, .poa, .kovan, .sokol, .callisto, .goerli, .artis_sigma1, .artis_tau1, .binance_smart_chain_testnet, .ropsten, .rinkeby, .heco, .heco_testnet, .fantom_testnet, .avalanche_testnet, .mumbai_testnet, .custom, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli, .cronosMainnet:
-            return nil
+        case .avalanche, .xDai, .classic, .fantom, .arbitrum, .candle, .polygon, .binance_smart_chain, .klaytnCypress, .klaytnBaobabTestnet, .poa, .kovan, .sokol, .callisto, .goerli, .artis_sigma1, .artis_tau1, .binance_smart_chain_testnet, .ropsten, .rinkeby, .heco, .heco_testnet, .fantom_testnet, .avalanche_testnet, .mumbai_testnet, .custom, .optimistic, .optimisticKovan, .cronosTestnet, .palm, .palmTestnet, .arbitrumRinkeby, .phi, .phi2, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli, .cronosMainnet: return nil
         }
     }
 
     var shouldExcludeZeroGasPrice: Bool {
         switch self {
-        case .klaytnCypress, .klaytnBaobabTestnet:
-            return true
-        case .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .optimistic, .polygon, .mumbai_testnet, .cronosTestnet, .arbitrum, .arbitrumRinkeby, .main, .kovan, .ropsten, .rinkeby, .poa, .classic, .callisto, .phi, .phi2, .goerli, .artis_sigma1, .artis_tau1, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .optimisticKovan, .sokol, .custom, .palm, .palmTestnet, .ioTeX, .ioTeXTestnet, .xDai, .candle, .optimismGoerli, .arbitrumGoerli, .cronosMainnet:
-            return false
+        case .klaytnCypress, .klaytnBaobabTestnet: return true
+        case .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .optimistic, .polygon, .mumbai_testnet, .cronosTestnet, .arbitrum, .arbitrumRinkeby, .main, .kovan, .ropsten, .rinkeby, .poa, .classic, .callisto, .phi, .phi2, .goerli, .artis_sigma1, .artis_tau1, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .optimisticKovan, .sokol, .custom, .palm, .palmTestnet, .ioTeX, .ioTeXTestnet, .xDai, .candle, .optimismGoerli, .arbitrumGoerli, .cronosMainnet: return false
         }
     }
 
     var rpcNodeBatchSupport: RpcNodeBatchSupport {
         switch self {
-        case .klaytnCypress, .klaytnBaobabTestnet:
-            return .noBatching
-        case .xDai:
-            return .batch(6)
-	case .cronosMainnet:
-            return .batch(5)
-        case .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .optimistic, .candle, .polygon, .mumbai_testnet, .cronosTestnet, .arbitrum, .arbitrumRinkeby, .main, .kovan, .ropsten, .rinkeby, .poa, .classic, .callisto, .phi, .phi2, .goerli, .artis_sigma1, .artis_tau1, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .optimisticKovan, .sokol, .custom, .palm, .palmTestnet, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli:
-            return .batch(32)
+        case .klaytnCypress, .klaytnBaobabTestnet: return .noBatching
+        case .xDai: return .batch(6)
+        case .cronosMainnet: return .batch(5)
+        case .binance_smart_chain, .binance_smart_chain_testnet, .heco, .heco_testnet, .optimistic, .candle, .polygon, .mumbai_testnet, .cronosTestnet, .arbitrum, .arbitrumRinkeby, .main, .kovan, .ropsten, .rinkeby, .poa, .classic, .callisto, .phi, .phi2, .goerli, .artis_sigma1, .artis_tau1, .fantom, .fantom_testnet, .avalanche, .avalanche_testnet, .optimisticKovan, .sokol, .custom, .palm, .palmTestnet, .ioTeX, .ioTeXTestnet, .optimismGoerli, .arbitrumGoerli: return .batch(32)
         }
     }
 

--- a/modules/AlphaWalletWeb3/AlphaWalletWeb3/Utils/CcipRead.swift
+++ b/modules/AlphaWalletWeb3/AlphaWalletWeb3/Utils/CcipRead.swift
@@ -20,15 +20,15 @@ class CcipRead {
     private let extraData: String
 
     init?(web3: Web3, options: Web3Options, onBlock: String, fromDataString dataString: String?) {
-        if let dataString = dataString, let (urls, sender, callbackSelector, callData, extraData) = Self.extractCcipRead(fromDataString: dataString) {
+        if let dataString = dataString, let ccipRead = Self.extractCcipRead(fromDataString: dataString) {
             self.web3 = web3
             self.options = options
             self.onBlock = onBlock
-            self.urls = urls
-            self.sender = sender
-            self.callbackSelector = callbackSelector
-            self.callData = callData
-            self.extraData = extraData
+            self.urls = ccipRead.urls
+            self.sender = ccipRead.sender
+            self.callbackSelector = ccipRead.callbackSelector
+            self.callData = ccipRead.callData
+            self.extraData = ccipRead.extraData
         } else {
             return nil
         }
@@ -114,15 +114,15 @@ class CcipRead {
     }
 
     //Must not convert `urls` to `[URL]` since URLs can't contain "{sender}" and "{data}"
-    private static func extractCcipRead(fromDataString dataString: String?) -> (urls: [String], sender: EthereumAddress, callbackSelector: String, callData: String, extraData: String)? {
+    private static func extractCcipRead(fromDataString dataString: String?) -> Ccip<EthereumAddress>? {
         guard let dataString = dataString?.addHexPrefix() else { return nil }
         //OffchainLookup(address sender, string[] urls, bytes callData, bytes4 callbackFunction, bytes extraData)
         let hashInterfaceForOffChainLookup = "0x556f1830"
         //count - 2 to exclude "0x"
         if dataString.hasPrefix(hashInterfaceForOffChainLookup), (dataString.count - 2) / 2 % 32 == 4 {
-            let (urls, sender, callbackSelector, callData, extraData) = _extractCcipRead(dataString: dataString)
-            if let sender = EthereumAddress(sender.addHexPrefix(), ignoreChecksum: true) {
-                return (urls, sender, callbackSelector, callData, extraData)
+            let ccipRead = _extractCcipRead(dataString: dataString)
+            if let sender = EthereumAddress(ccipRead.sender.addHexPrefix(), ignoreChecksum: true) {
+                return .init(urls: ccipRead.urls, sender: sender, callbackSelector: ccipRead.callbackSelector, callData: ccipRead.callData, extraData: ccipRead.extraData)
             } else {
                 return nil
             }
@@ -131,7 +131,15 @@ class CcipRead {
         }
     }
 
-    private static func _extractCcipRead(dataString: String) -> (urls: [String], sender: String, callbackSelector: String, callData: String, extraData: String) {
+    private struct Ccip<T> {
+        let urls: [String]
+        let sender: T
+        let callbackSelector: String
+        let callData: String
+        let extraData: String
+    }
+
+    private static func _extractCcipRead(dataString: String) -> Ccip<String> {
         let dataString: String = {
             //8 characters for 4 bytes interface hash and +2 for "0x"
             let result = String(dataString.dropFirst(8 + 2))
@@ -180,7 +188,7 @@ class CcipRead {
 
         let extraData = parseBytes(data: dataString, start: 128*2)
 
-        return (urls.compactMap { $0 }, sender, callbackSelector, callData, extraData)
+        return .init(urls: urls.compactMap { $0 }, sender: sender, callbackSelector: callbackSelector, callData: callData, extraData: extraData)
     }
 
     private static func buildCcipJsonRpcCallbackPayloadHexString(callbackSelector: String, ccipGatewayCallResult ccipGatewayCallResultRaw: String, extraData: String) -> String {

